src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -131,6 +129,23 @@ private async Task ProduceDiagnosticsAsync(
                         AddIncludedDiagnostics(builder, analysisResult.GetOtherDiagnostics());
                     }
                 }
+
+                async Task<ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>> GetOrComputeDiagnosticAnalysisResultsAsync(ImmutableArray<StateSet> stateSets)
+                {
+                    // If there was a 'ForceAnalyzeProjectAsync' run for this project, we can piggy back off of the
+                    // prior computed/cached results as they will be a superset of the results we want.
+                    //
+                    // Note: the caller will loop over *its* state sets, grabbing from the full set of data we've cached
+                    // for this project, and filtering down further.  So it's ok to return this potentially larger set.
+                    if (this.Owner._projectToForceAnalysisData.TryGetValue(project, out var box))
+                        return box.Value.diagnosticAnalysisResults;
+
+                    // Otherwise, just compute for the state sets we care about.
+                    var compilation = await GetOrCreateCompilationWithAnalyzersAsync(project, stateSets, Owner.AnalyzerService.CrashOnAnalyzerException, cancellationToken).ConfigureAwait(false);
```

Comment:
calls GetOrCreateCompilationWithAnalyzersAsync instead of CreateCompilationXXX.  We want this to reuse a cached value if applicable.